### ISSUE TITLE
feat: add `Store<bytes31>` implementation

### ISSUE THREAD

**Issue by enitrat** [starkware-libs/cairo#4256]
# Feature Request

**Describe the Feature Request**
Add an implementation of `Store<bytes31>` in `starknet/storage_access.cairo`. I will implement it if accepted

**Describe Preferred Solution**
```rust

impl StoreBytes31 of Store<bytes31> {
    fn read(address_domain: u32, base: StorageBaseAddress) -> SyscallResult<bytes31> {
        Result::Ok(
            Store::<felt252>::read(address_domain, base)?
                .try_into()
                .expect('StoreBytes31 - non bytes31')
        )
    }
    #[inline(always)]
    fn write(address_domain: u32, base: StorageBaseAddress, value: bytes31) -> SyscallResult<()> {
        Store::<felt252>::write(address_domain, base, value.into())
    }
    #[inline(always)]
    fn read_at_offset(
        address_domain: u32, base: StorageBaseAddress, offset: u8
    ) -> SyscallResult<bytes31> {
        Result::Ok(
            Store::<felt252>::read_at_offset(address_domain, base, offset)?
                .try_into()
                .expect('StoreBytes31 - non bytes31')
        )
    }
    #[inline(always)]
    fn write_at_offset(
        address_domain: u32, base: StorageBaseAddress, offset: u8, value: bytes31
    ) -> SyscallResult<()> {
        Store::<felt252>::write_at_offset(address_domain, base, offset, value.into())
    }
    #[inline(always)]
    fn size() -> u8 {
        1_u8
    }
}

```

**Describe Alternatives**

**Related Code**

**Additional Context**

**If the feature request is approved, would you be willing to submit a PR?**
_(Help can be provided if you need assistance submitting a PR)_

- [ ] Yes
- [ ] No

**Comment by rcatalan98** [comment]
Hi,
This feature would be really useful! In addition to this one, a `Store< ByteArray` would come handy as well. We would use this for a side project to [mint pictures fully on-chain](https://github.com/GrindLabsOrg/NFT-onchain). 
Happy to help if it's necessary!

**Comment by enitrat** [comment]
> Hi, This feature would be really useful! In addition to this one, a `Store< ByteArray` would come handy as well. We would use this for a side project to [mint pictures fully on-chain](https://github.com/GrindLabsOrg/NFT-onchain). Happy to help if it's necessary!

For ByteArray I think the best is to store the `data` field in a `List`, and the `pending_word` and `pending_word_len` in tw o standalone storage variables. 

Then you can easily reconstruct the ByteArray struct by loading these three from storage.

**Comment by orizi** [comment]
I agree with @enitrat - generally we should probably have another trait for `storage proxy` or something similar to enable recursive storage of generic types.
This is a thing we are considering in general for additional contract features.

**Comment by orizi** [comment]
Closed by #4280
